app_dart/lib/src/request_handlers/reset_prod_task.dart

```diff
@@ -82,20 +97,24 @@ class ResetProdTask extends ApiRequestHandler<Body> {
       },
       orElse: () => null,
     );
-    log.info('Owner: $owner, Repo: $repo, Builder: $builder, CommitSha: ${commit.sha}, Build: $build');
+    log.info('Owner: $owner, Repo: $repo, Builder: $builder, CommitSha: $commitSha, Build: $build');
 
     if (build != null) {
       throw const ConflictException();
     }
     await luciBuildService.rescheduleProdBuild(
-      commitSha: commit.sha,
+      commitSha: commitSha,
       builderName: builder,
+      repo: repo,
     );
-    task
-      ..status = Task.statusNew
-      ..startTimestamp = 0
-      ..attempts += 1;
-    await datastore.insert(<Task>[task]);
+    if (task != null) {
```

Comment:
The logic allows retry framework build via curl as well, for which case the task is null and will not be updated in the datastore. Shall we enforce engine only, and restricting the framework retry to dashboard only?

Reply:
Forced an exception if we try to use curl with flutter repo.